### DISCUSSION TITLE
Best way to extract a tunnel "--rsd HOST PORT" and pass to a new process?

### DISCUSSION THREAD

**Discussion by blakegonzales** [doronz88/pymobiledevice3#615] [Q&A] [answered]
I'm curious if anyone has a better way to extract the tunnel "--rsd HOST PORT" and pass it to a new process.  Preferably, I would like to do this in one script instead of two.  Here is how I do it today:

I first run this script to open then tunnel.

    #/bin/bash
    TEMPFILE=/tmp/simulate-location-rsd.txt
    sudo script -q -F $TEMPFILE pymobiledevice3 remote start-quic-tunnel --script-mode

Then I open up **another** terminal window and run this script:

    #/bin/bash
    TEMPFILE=/tmp/simulate-location-rsd.txt
    pymobiledevice3 developer dvt simulate-location set --rsd `cat $TEMPFILE | cut -w -f1,2` -- $1 $2

Any thoughts from the community?

**Comment by martindasilva404** [comment]


**Reply by netanelc305** [reply]
Sorry, I accidentally deleted your message 🤦‍♂️

**Comment by netanelc305** [comment]
> I'm curious if anyone has a better way to extract the tunnel "--rsd HOST PORT" and pass it to a new process.  Preferably, I would like to do this in one script instead of two.  Here is how I do it today:
> 
> I first run this script to open then tunnel.
> 
>     #/bin/bash
>     TEMPFILE=/tmp/simulate-location-rsd.txt
>     sudo script -q -F $TEMPFILE pymobiledevice3 remote start-quic-tunnel --script-mode
> 
> Then I open up **another** terminal window and run this script:
> 
>     #/bin/bash
>     TEMPFILE=/tmp/simulate-location-rsd.txt
>     pymobiledevice3 developer dvt simulate-location set --rsd `cat $TEMPFILE | cut -w -f1,2` -- $1 $2
> 
> Any thoughts from the community?


Hi,
Check out the new feature `Tunneld`.

**Reply by blakegonzales** [reply]
`tunneld` looks promising!

I'm not sure of the usage though.  I'm trying pseudo-random `--host` and `--port` and getting errors.

```
% sudo pymobiledevice3 remote tunneld --host fd31:226:df0b::1 --port 61532
INFO:     Started server process [58501]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
ERROR:    [Errno 49] error while attempting to bind on address ('fd31:226:df0b::1', 61532, 0, 0): can't assign requested address
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
%
```

What is the correct usage?

**Reply by netanelc305** [reply, marked as answer]
The `--host` and `--port` options are intended for `Tunneld` http server, which can be used to query information about active tunnels. It's generally recommended to use the default values for these options.

Start Tunneld:
```bash
sudo pymobiledevice3 remote tunneld -d
```

Once you've started the `Tunneld` service, you can use various commands that require the `--rsd` flag, like this:
```bash
pymobiledevice3 developer dvt simulate-location --tunnel ''
```

Please note that this feature is currently not supported on Linux and is compatible only with macOS.

**Reply by blakegonzales** [reply]
Amazing.  That works perfectly and is so much easier!

After starting the tunnel as shown, I ran the following command.

`pymobiledevice3 developer dvt simulate-location set --tunnel '' -- [LATITUDE] [LONGITUDE]
`

Thank you!

**Reply by netanelc305** [reply]
You're welcome! If you have any more questions or need further assistance, feel free to ask here or on our [discord](https://discord.gg/52mZGC3JXJ) .  😄
